### PR TITLE
feat(dependabot): private registry support

### PR DESCRIPTION
**Pull request by kcwinner**
https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/configuration-options-for-dependency-updates#configuration-options-for-private-registries

Example:
```
registries:
  npm-ghp-octocat:
    type: npm-registry
    url: https://npm.pkg.github.com
    token: ${{secrets.TOKEN}}

updates:
    package-ecosystem: npm
    directory: "/"
    registries: 
      - npm-ghp-octocat
    schedule:
      interval: daily
```
---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/github/workflows.ts [hunk 1]
```diff
@@ -73,7 +73,7 @@ function snakeCaseKeys<T = unknown>(obj: T): T {
   return result as any;
 }
 
-function kebabCaseKeys<T = unknown>(obj: T, recursive = true): T {
+export function kebabCaseKeys<T = unknown>(obj: T, recursive = true): T {
```
Comment (kcwinner) [inline src/github/workflows.ts:76]: This was iffy. Wondering if it should be pulled out/up to be re-used.

File: src/github/dependabot.ts [hunk 2]
```diff
@@ -43,6 +44,60 @@ export interface DependabotOptions {
    * List of labels to apply to the created PR's.
    */
   readonly labels?: string[];
+
+  /**
+   * Map of dependabot registries to use
+   * @default undefined
```
Comment (eladb) [inline src/github/dependabot.ts:50]:
```suggestion
   * @default - use public registries
```

File: src/github/dependabot.ts [hunk 3]
```diff
@@ -43,6 +44,60 @@ export interface DependabotOptions {
...
+   */
+  readonly registries?: { [name: string]: DependabotRegistry };
+}
+
+/**
+ * Use to add private registry support for dependabot
+ * https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/configuration-options-for-dependency-updates#configuration-options-for-private-registries
```
Comment (eladb) [inline src/github/dependabot.ts:57]:
```suggestion
 * @see https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/configuration-options-for-dependency-updates#configuration-options-for-private-registries
```

File: src/github/dependabot.ts [hunk 4]
```diff
@@ -43,6 +44,60 @@ export interface DependabotOptions {
...
+ */
+export interface DependabotRegistry {
+  /**
+   * Registry type e.g. 'npm-registry' or 'docker-registry'
+   */
+  readonly type: string;
```
Comment (eladb) [inline src/github/dependabot.ts:63]: enum?

File: src/github/dependabot.ts [hunk 5]
```diff
@@ -43,6 +44,60 @@ export interface DependabotOptions {
...
+
+  /**
+   * Url for the registry e.g. 'https://npm.pkg.github.com' or 'registry.hub.docker.com'
+   */
+  readonly url: string;
+
+  /**
+   * The username that Dependabot uses to access the registry
+   * @default undefined
```
Comment (eladb) [inline src/github/dependabot.ts:72]: In such cases we prefer to specify the default behavior (i.e. what `undefined` means):

```suggestion
   * @default - do not authenticate
```

File: src/github/workflows.ts (hunk 1 above)
Comment (eladb) [reply src/github/workflows.ts:76]: Pull out into `util.ts` please which is not exported out.

**Review by eladb — CHANGES_REQUESTED** [review]

**Comment by kcwinner** [comment]
Been slow due to being on my honeymoon! Addressed the comments and pulled the registry types into an enum

**Review by eladb — APPROVED** [review]

**Pull request merged**